### PR TITLE
Option to terminate visibility timeout on error

### PR DESCRIPTION
**Pull request by rocketspacer**
As mentioned in #72

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -162,6 +163,19 @@ Consumer.prototype._processMessage = function (message, cb) {
       } else {
         consumer.emit('processing_error', err, message);
       }
+
+      if (consumer.terminateVisibilityTimeout) {
```
Comment (nspragg) [inline index.js]: We aim to maintain 100% test coverage, so we'll need some tests for this new functionality. To check test coverage run `npm run coverage`. It's also worth running the linter too. 

For more details see:
https://github.com/bbc/sqs-consumer/blob/master/CONTRIBUTING.md

File: index.js [hunk 2]
```diff
@@ -162,6 +163,19 @@ Consumer.prototype._processMessage = function (message, cb) {
...
+        consumer.sqs.changeMessageVisibility({
+          QueueUrl: consumer.queueUrl,
+          ReceiptHandle: message.ReceiptHandle,
+          VisibilityTimeout: 0
+        }, function(err) {
+          if (err)
```
Comment (nspragg) [inline index.js]: Just a style comment, but please put consumer on the same line as the if or add braces.

**Comment by rocketspacer** [comment]
Added tests

File: test/index.js [hunk 3]
```diff
@@ -359,6 +361,51 @@ describe('Consumer', function () {
 
       consumer.start();
     });
+
+    it('terminate message visibility timeout on processing error', function(done) {
+      handleMessage.yields(new Error('Processing error'));
+
+      // Turn on terminate visibility timeout option
+      consumer.terminateVisibilityTimeout = true;
```
Comment (nspragg) [inline test/index.js]: Thanks for adding some tests. I think we need one more test to ensure that `sqs.changeMessageVisibility` is not called when `terminateVisibilityTimeout` is set to false.

Comment (nspragg) [inline test/index.js]: As for the release - hopefully within 24 hours

**Comment by rocketspacer** [comment]
Updated

**Pull request merged**